internal/service/seller.go

```diff
@@ -13,12 +15,60 @@ type SellersService struct {
 	rp repository.SellersRepository
 }
 
-func (s *SellersService) GetAll() (v map[int]model.Seller, err error) {
-	v, err = s.rp.Get()
+//Validates
+func (s *SellersService) ValidateFields(seller model.Seller) error {
```

Comment:
talvez assim seria mais escalavel só uma sugestão isso não está errado mas se for um validacao maior seria interessante

func (s *SellersService) ValidateFields(seller model.Seller) error {
	fieldsToValidate := []struct {
		value interface{}
		validateFunc func(interface{}) error
	}{
		{seller.CompanyName, validateFormatString},
		{seller.Address, validateFormatString},
		{seller.Telephone, validateFormatString},
		{seller.CID, validateFormatInt},
	}

	for _, field := range fieldsToValidate {
		if err := field.validateFunc(field.value); err != nil {
			return err
		}
	}

	return nil
}